### PR TITLE
Fixes #23973: 

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23973

### CODE REVIEW

**Comment by Fdall** [comment]
This is a draft

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
PR updated with a new commit

File: policies/rudderc/tests/cases/general/ntp/technique.ps1 [hunk 1]
```diff
@@ -24,59 +24,30 @@
...
         ComponentKey = $componentKey
-        ComponentName = "NTP service"
+        ComponentName = 'NTP service'
         PolicyMode = $policyMode
         ReportId = $reportId
         DisableReporting = $false
         TechniqueName = $techniqueName
     }
-    
-    $class = "linux.fedora"
```
Comment (amousset) [inline policies/rudderc/tests/cases/general/ntp/technique.ps1:68]: this is quite unexpected that this class disapers if the source was not modified

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
PR rebased

File: policies/rudderc/tests/cases/general/ntp/technique.ps1 [hunk 2]
```diff
@@ -39,10 +39,10 @@
     if ($localContext.Evaluate($class)) {
         $methodParams = @{
             Architecture = ""
-            Name = "htop"
+            Name = "'htop'"
             Provider = ""
             Version = @'
-2.3.4
+'2.3.4'
```
Comment (Fdall) [inline policies/rudderc/tests/cases/general/ntp/technique.ps1:45]: this is not correct

File: policies/rudderc/tests/cases/general/ntp/technique.ps1 [hunk 3]
```diff
@@ -39,10 +39,10 @@
     if ($localContext.Evaluate($class)) {
         $methodParams = @{
             Architecture = ""
-            Name = "htop"
+            Name = "'htop'"
```
Comment (Fdall) [inline policies/rudderc/tests/cases/general/ntp/technique.ps1:42]: this is incorrect

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**